pkg/project/project_test.go

```diff
@@ -12,32 +11,42 @@ import (
 	ktesting "k8s.io/client-go/testing"
 )
 
-
 func TestDelete(t *testing.T) {
 	tests := []struct {
-		name        string
-		wantErr     bool
-		projectName string
+		name                             string
+		wantErr                          bool
+		wantMockSetCurrentProjectInvoked bool
+		projectName                      string
 	}{
 		{
-			name:        "Test project delete for multiple projects",
-			wantErr:     false,
-			projectName: "prj2",
+			name:    "Test project delete for multiple projects",
+			wantErr: false,
+			wantMockSetCurrentProjectInvoked: true,
+			projectName:                      "prj2",
 		},
 		{
-			name:        "Test only project delete",
-			wantErr:     false,
-			projectName: "prj1",
+			name:    "Test delete the only remaining project",
+			wantErr: false,
+			wantMockSetCurrentProjectInvoked: false,
+			projectName:                      "testing",
 		},
 	}
 
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
 
+			mockSetCurrentProjectInvoked := false
 			// Fake the client with the appropriate arguments
 			client, fakeClientSet := occlient.FakeNew()
+			occlient.SetCurrentProject = func(project string, c *occlient.Client) error {
+				mockSetCurrentProjectInvoked = true
+				return nil
+			}
```

Comment:
I don't think we need this. We need to test SetCurrentProject separately. Whether a project is set as active or not, we could have that cover under e2e tests.

Reply:
@ashetty1 Done updated PR accordingly. Please have a look